docs/site-generator/Makefile

```diff
@@ -0,0 +1,155 @@
+# Default shell
+SHELL := /bin/bash
+
+# Config
+SHARED := $(shared)
+export BASEURL := $(baseurl)
+export OUTPUT := $(output)
+export OWNER := $(owner)
+export REPOSITORY := $(repository)
+export VERSION := $(version)
+export LANGUAGES := $(languages)
+
+URL_TARBALL := https://github.com/$(OWNER)/$(REPOSITORY)/archive/$(VERSION).tar.gz
+REPO_TMP_PATH := $(SHARED)/$(OWNER)-$(REPOSITORY)-$(VERSION)
```

Comment:
I focused in making the full thing fully working with the current CI.
I'd focus in caching and reusing shared in a next iteration if you agree with the main structure